sqlglot/dialects/tsql.py

```diff
@@ -296,6 +297,14 @@ class Parser(parser.Parser):
             "SYSDATETIME": exp.CurrentTimestamp.from_arg_list,
             "SUSER_NAME": exp.CurrentUser.from_arg_list,
             "SUSER_SNAME": exp.CurrentUser.from_arg_list,
+            "SYSTEM_USER": exp.CurrentUser.from_arg_list,
+        }
+
+        JOIN_HINTS = {
+            TokenType.LOOP,
+            TokenType.HASH,
+            TokenType.MERGE,
+            TokenType.REMOTE,
         }
```

Comment:
Maybe we should use a set of strings here instead? Nowadays we try to not create new tokens for things like `LOOP` and just use `self._match_texts`.

Reply:
Thanks, this is a bit more handy in this case indeed.